src/storage/src/source/reclock.rs

```diff
@@ -559,12 +559,22 @@ where
                 upper = Antichain::new();
             }
 
+            // If this is the first binding we mint then we will mint it at the minimum target
+            // timestamp. The first source upper is always the upper of the snapshot and by mapping
+            // it to the minimum target timestamp we make it so that the final shard never appears
+            // empty at any timestamp.
```

Comment:
this will add 1 second of latency to the the snapshot being actually queryable, right? because we will need the next batch to bump it to a more recent time? Not saying we have to change this, but worth a note

Reply:
Hm, I don't think so. Where does that extra second come from? If the snapshot happened at `offset_snapshot` then the initial progress update we'll get (and reclock to timestmap 0) will be something like `{offset_snapshot + 1}`. Are you thinking something else?

Reply:
but the snapshot's ts will be `0`, right? so a query right after the snapshot whose query ts is ~now() will have to wait, right?

Reply:
The snapshot ts will be `0` but the first frontier advancement will be `[1]`, so it will be available immediately

Reply:
you would need to `select ... AS OF 0` for that, right?

Reply:
I misspoke. The **binding_ts** is `0` but the frontier still advances to `{now()}`, so you immediately make the collection querable just like before this PR and the only difference is where the first binding lands. Previously it would land at `now() - 1` and now it lands at `0`.

Reply:
ah i see, i forgot that the binding time is not used as the frontier, its the actual shard frontier that is